src/WCSimWCSD.cc

```diff
@@ -163,11 +164,16 @@ G4bool WCSimWCSD::ProcessHits(G4Step* aStep, G4TouchableHistory*)
      G4double local_y = localPosition.y();
      G4double local_z = localPosition.z();
      theta_angle = acos(fabs(local_z)/sqrt(pow(local_x,2)+pow(local_y,2)+pow(local_z,2)))/3.1415926*180.;
-     effectiveAngularEfficiency = fdet->GetPMTCollectionEfficiency(theta_angle, collectionName[0]);
+     effectiveAngularEfficiency = fdet->GetPMTCollectionEfficiency(theta_angle, volumeName);
      if (G4UniformRand() <= effectiveAngularEfficiency || fdet->UsePMT_Coll_Eff()==0){
-
+       //Retrieve the pointer to the appropriate hit collection. Since volumeName is the same as the SD name, this works. 
```

Comment:
Is this a G4 requirement, or is this just a convention of our code? If it is just a convention, perhaps we need to do something more robust, or at least put a comment where the SD and volume get named that makes it clear that this is required. It actually looks like the collection name, SD name, and volume name are all the same. Is that the case?